core/src/main/kotlin/net/corda/core/crypto/Crypto.kt

```
@@ -0,0 +1,82 @@
+    @Throws(CryptoException::class)
+    fun generateKeyPair(algorithmName: String): KeyPair {
+        return findAlgorithm(algorithmName)?.keyPairGenerator?.generateKeyPair() ?: throw CryptoException("Unsupported algorithm during key generation for String: $algorithmName")
+    }
+
+    /**
+     * Generic way to sign [ByteArray] messages with a [PrivateKey]. Strategy on the actual signing algorithm is based
+     * on the [PrivateKey] type. This class has similarities to the [sign] but it does not attach the public key information.
+     *
+     * @param privKey the signer's [PrivateKey].
+     * @param bytesToSign the message to be signed in [ByteArray] form.
+     * @return the digital signature on the input message.
+     * @throws CryptoException if privKey algorithm-type is not supported.
+     * @throws InvalidKeyException if the key is invalid.
+     * @throws SignatureException if signing is not possible due to malformed data or private key.
+     */
+    @Throws(CryptoException::class, InvalidKeyException::class, SignatureException::class)
+    fun doSign(privKey: PrivateKey, bytesToSign: ByteArray): ByteArray {
+        val sig: Signature = findAlgorithm(privKey)?.sig ?: throw CryptoException("Unsupported key/algorithm during signing (privKey: ${privKey})")
+        sig.initSign(privKey)
+        sig.update(bytesToSign)
+        return sig.sign()
+    }
+
+    /**
+     * Utility to simplify the act of verifying a signature.
+     *
+     * @throws CryptoException if privKey algorithm-type is not supported.
+     * @throws InvalidKeyException if the key is invalid.
+     * @throws SignatureException if this signature object is not initialized properly,
+     * the passed-in signature is improperly encoded or of the wrong type,
+     * if this signature algorithm is unable to process the input data provided, etc.
+     */
+    @Throws(CryptoException::class)
+    fun doVerify(pubKey: PublicKey, signature: ByteArray, clearData: ByteArray): Boolean {
```

I much prefer signature checking functions to throw exceptions rather than return false. That way if the caller forgets to check at least there's a chance it'll fail noisily instead of silently.